### PR TITLE
Add downlink count for end devices in the Console

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/issues/2740

<details>
<summary>Screenshots</summary>
Screenshot 2020-09-28 at 15 31 04
Screenshot 2020-09-28 at 15 31 40

</details>

#### Changes

- Extract downlink frame count from the `ns.down.data.schedule.attempt` event.
- Adjust styles for `<MessagesCount />`

#### Testing

Tested with actual devices on ch staging

#### Notes for Reviewers

@rvolosatovs Please confirm that the `ns.down.data.schedule.attempt` event is the one to check for downlink count. Also, can we use this (or any other) event to bump `last seen` timer for an end device in the console related to downlinks? Currently, we do this only for uplinks (`ns.up.data.receive`, `ns.up.join.receive`, `ns.up.rejoin.receive`)

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by rvolosatovs** [comment]
> @rvolosatovs Please confirm that the `ns.down.data.schedule.attempt` event is the one to check for downlink count.

It is indeed the event you should be using, however there are 2 distinct downlink counters.
Pre 1.1 there's only one and only one counter should be displayed.
In 1.1 and later versions - there are 2 counters. If `FPort` in downlink is equal to `0` - then it's the "network" frame counter, otherwise it's the "application" frame counter. Both counters have to be displayed.

> Also, can we use this (or any other) event to bump last seen timer for an end device in the console related to downlinks? Currently, we do this only for uplinks (ns.up.data.receive, ns.up.join.receive, ns.up.rejoin.receive)

No, you can only update last seen on uplinks.

File: pkg/webui/console/components/entity-title-section/content/messages-count/messages-count.styl [hunk 1]
```diff
@@ -14,3 +14,6 @@
 
 .container
   margin-left: $ls.s
+
+  &:not(:first-of-type)
+    margin-left: $ls.xxs
```
Comment (kschiffer) [inline pkg/webui/console/components/entity-title-section/content/messages-count/messages-count.styl:19]:
```suggestion
.container:first-of-type
    margin-left: $ls.s

.container:not(:first-of-type)
    margin-left: $ls.xxs
```

File: pkg/webui/console/store/reducers/devices.js [hunk 2]
```diff
@@ -120,10 +121,13 @@ const devices = function(state = defaultState, { type, payload, event }) {
 
       // Detect uplink process event to update uplink frame count state.
```
Comment (kschiffer) [inline pkg/webui/console/store/reducers/devices.js]:
```suggestion
      // Detect uplink/downlink process events to update uplink/downlink frame count state.
```

**Review by kschiffer — APPROVED** [review]

**Comment by bafonins** [comment]
> In 1.1 and later versions - there are 2 counters. If FPort in downlink is equal to 0 - then it's the "network" frame counter, otherwise it's the "application" frame counter. Both counters have to be displayed.

I suggest we stick to the current implementation, but display only the "application" frame counter for 1.1+ devices by cheking the `f_port` value. @rvolosatovs What do you think?

The problem is that I cant really test this with a 1.1 device. I can file an issue for future adjustments.

**Comment by rvolosatovs** [comment]
> I suggest we stick to the current implementation, but display only the "application" frame counter for 1.1+ devices by cheking the `f_port` value. @rvolosatovs What do you think?

SGTM, that's the more relevant one of the two from the perspective of the user.
The other one is mostly useful for debugging of the device operation.
Please file a backlog issue about showing it as well.

**Comment by bafonins** [comment]
@rvolosatovs Is there any way we can distinguish between `< 1.1` and `>= 1.1` end devices by just looking at the event payload?

**Comment by rvolosatovs** [comment]
> @rvolosatovs Is there any way we can distinguish between `< 1.1` and `>= 1.1` end devices by just looking at the event payload?

Depends on the event, but general answer is "no".
In 1.1 there are MAC commands(`Rekey` and `Reset`) that change the MAC version of the device within a session. Console will have to update the MAC version in response to these after the initial GET of the device.
Perhaps, we could introduce a "MAC version change" event, similar to how we have one for class change.
We should also probably have a `session start` event, which would be triggered on first uplink confirming the session or on creation for ABP, that would contain MAC version among other things. @johanstokking what do you think?

**Comment by johanstokking** [comment]
> We should also probably have a `session start` event, which would be triggered on first uplink confirming the session or on creation for ABP, that would contain MAC version among other things

This makes the event stream stateful and I don't think we should do that. If you want to differentiate the display per LoRaWAN version, which makes perfect sense and aligns with what we already do elsewhere, it should be from fetching the end device.

**Pull request merged**